test/e2e/util_test.go

```diff
@@ -179,6 +205,62 @@ func waitForDelete(checkResource checkResourceFunc) error {
 	return err
 }
 
+type catalogSourceCheckFunc func(*v1alpha1.CatalogSource) bool
+
+func catalogSourceSynced(catalog *v1alpha1.CatalogSource) bool {
+	if !catalog.Status.LastSync.IsZero() {
+		return true
+	}
+	return false
+}
+
+func fetchCatalogSource(t *testing.T, crc versioned.Interface, name, namespace string, check catalogSourceCheckFunc) (*v1alpha1.CatalogSource, error) {
+	var fetched *v1alpha1.CatalogSource
+	var err error
+
+	err = wait.Poll(pollInterval, pollDuration, func() (bool, error) {
+		fetched, err = crc.OperatorsV1alpha1().CatalogSources(namespace).Get(name, metav1.GetOptions{})
+		if err != nil || fetched == nil {
+			return false, err
+		}
+		return check(fetched), nil
+	})
+
+	return fetched, err
+}
+
+func createFieldNotEqualSelector(field string, names ...string) string {
+	var builder strings.Builder
+	for i, name := range names {
+		builder.WriteString(field)
+		builder.WriteString("!=")
+		builder.WriteString(name)
+		if i < len(names)-1 {
+			builder.WriteString(",")
+		}
+	}
+
+	return builder.String()
+}
+
+func cleanupOLM(t *testing.T, namespace string) {
```

Comment:
the frustrating thing about using this everywhere is that it clears out the state of the test when you `defer` a call to it. it sort of defeats the purpose of keeping the test namespace around when the tests fail (there's nothing to inspect).

Reply:
Makes total sense.